### PR TITLE
runtime(java): Optionally highlight the :: token for method references

### PR DESCRIPTION
**Pull request by zzzyxwvut**
This token will be highlighted, similar to the arrow of  
lambda expressions, whenever `g:java_highlight_functions` is  
defined.

Also:

- Improve the recognition of _switch-case_ labels  
  (D-Cysteine).
- Remove insignificant empty statements in syntax test  
  files.

References:
https://docs.oracle.com/javase/specs/jls/se21/html/jls-15.html#jls-15.13  
https://github.com/fleiner/vim/pull/1

### CODE REVIEW

File: runtime/syntax/java.vim [hunk 1]
```diff
@@ -260,7 +260,7 @@ if exists("java_space_errors")
   endif
 endif
 
-exec 'syn match javaUserLabel "^\s*\<\K\k*\>\%(\<default\>\)\@' . s:ff.Peek('7', '') . '<!\s*:"he=e-1'
+exec 'syn match javaUserLabel "^\s*\<\K\k*\>\%(\<default\>\)\@' . s:ff.Peek('7', '') . '<!\s*:\(:\)\@!"he=e-1'
```
Comment (dkearns) [inline runtime/syntax/java.vim]:
```suggestion
exec 'syn match javaUserLabel "^\s*\<\K\k*\>\%(\<default\>\)\@' . s:ff.Peek('7', '') . '<!\s*::\@!"he=e-1'
```

Comment (dkearns) [inline runtime/syntax/java.vim]:
```suggestion
exec 'syn match javaUserLabel "^\s*\<\K\k*\>\%(\<default\>\)\@' . s:ff.Peek('7', '') . '<!\s*::\@!"he=e-1'
```

Comment (zzzyxwvut) [reply runtime/syntax/java.vim]: Oops!  Copy-pasting be damned.

**Comment by zzzyxwvut** [comment]
The suggested changes have been applied.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**